soramitsu/soramitsu-js-ui-library
# General project update

- Migrate from Yarn & Lerna to PNPM
- Use Storybook v7 (beta). It works with PNPM! But the stories should be re-written into [Component Story Format 3.0](https://storybook.js.org/blog/component-story-format-3-0). I only tested it on `SAlert` component so far.
- Use TypeScript 5 (release candidate). It highlighted some new errors...
- Tried to use much stricter TypeScript configuration, but we have 100+ errors with them! I left it for an upcoming PR.
- Update ESLint rules. I had to disable some of them, and I think it should be fixed in an upcoming PR.
- Remove Jest, use Vitest everywhere
- Remove API Extractor. Use `rollup-plugin-dts` for `.d.ts` bundle
- Set packages' type to `module`
- Update a lot of packages
- Use `^x.y` in prod dependencies for better (?) deps resolution experience of library users
- etc etc

This PR leaves a lot of TODOs and FIXMEs, and I think those should be fixed separately. Also I think it makes sense to release new versions with a pre-release tag, because there might be unexpected regressions.
